src/topology/continuous_function/algebra.lean

```diff
@@ -825,4 +826,31 @@ instance [has_star R] [has_star β] [has_smul R β] [star_module R β]
 
 end star_structure
 
+variables {X Y Z : Type*} [topological_space X] [topological_space Y] [topological_space Z]
+variables (𝕜 : Type*) [comm_semiring 𝕜]
+variables (A : Type*) [topological_space A] [semiring A] [topological_semiring A] [star_ring A]
+variables [has_continuous_star A] [algebra 𝕜 A]
+
+/-- The functorial map taking `f : C(X, Y)` to `C(Y, A) →⋆ₐ[A] C(X, A)` given by pre-composition
+with the continuous function `f`. -/
+def comp_star_alg_hom (f : C(X, Y)) : C(Y, A) →⋆ₐ[𝕜] C(X, A) :=
+{ to_fun := λ g, g.comp f,
+  map_one' := one_comp _,
+  map_mul' := λ _ _, rfl,
+  map_zero' := zero_comp _,
+  map_add' := λ _ _, rfl,
+  commutes' := λ _, rfl,
+  map_star' := λ _, rfl }
```

Comment:
Do we have any weaker bundlings of this elsewhere? This is true as a `monoid_hom` when `A` is just a monoid, etc; right?

I don't think you necessarily need to add all the intermediate bundings first, but it would be good to reference any similar bundlings we already have in its docstring.

Reply:
Yes, we have `continuous_map.comp_monoid_hom'` and `continuous_map.comp_add_monoid_hom'`, but I think these are the only ones. I'll mention them.

Reply:
If those names are primed, should this one also be?

Reply:
To be honest, I can't figure out why the other names are primed. I would favor not priming unless there is a good reason besides matching.